src/include/execution/persist/csv_reader.h

```diff
@@ -0,0 +1,43 @@
+//===----------------------------------------------------------------------===//
+// Copyright 2018-2023 Stichting DuckDB Foundation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice (including the next paragraph)
+// shall be included in all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <vector>
+
+#include "buffer/buffer_pool_manager.h"
+#include "catalog/catalog.h"
+#include "concurrency/transaction.h"
+#include "concurrency/transaction_manager.h"
+#include "execution/executor_context.h"
+#include "execution/executor_factory.h"
+#include "execution/executors/init_check_executor.h"
+#include "execution/plans/abstract_plan.h"
+#include "storage/table/tuple.h"
+
+namespace bustub {
+class BaseCsvReader {
+ public:
+  BaseCsvReader();
```

Comment:
no implementation yet? 🤪

Reply:
sorry, I still read the duckdb code and I'm a little slow to catch on that.